### PR TITLE
Don't accidentally create a folder owned by root

### PR DESCRIPTION
**Pull request by ala-ableton**

### CODE REVIEW

**Review by mst-ableton — COMMENTED** [review]
Generally looks ok, but why does

```
-v /jenkins/workspace:/ws
```

create a root-owned directory while

```
-v /jenkins/workspace:/jenkins/workspace:rw,
```

(which is provided by the Docker Pipeline Plugin) does not?

File: Dockerfile [hunk 1]
```diff
@@ -16,11 +16,8 @@ COPY pom.xml /opt/
 COPY ruleset.groovy /opt/resources/
 COPY run_codenarc.py /opt/
 
-WORKDIR /opt
-RUN python3.8 run_codenarc.py --resources /opt/resources
+RUN python3.8 /opt/run_codenarc.py --resources /opt/resources
 RUN groupadd -r jenkins && useradd --no-log-init -r -g jenkins jenkins
 USER jenkins
 
-WORKDIR /ws
```
Comment (mst-ableton) [inline Dockerfile:24]: Does this removing this line mean that users who run the container outside of Jenkins need to provide `-w`?

**Comment by ala-ableton** [comment]
The first sentence of the commit message body is "If the `env.WORKSPACE` folder doesn't exist". This is not the case in this repository, but it is the case in repositories where we download the `Jenkinsfile` from GitHub and execute it on the Jenkins built-in node (AKA controller).

File: Dockerfile (hunk 1 above)
Comment (ala-ableton) [reply Dockerfile:24]: Good point... Maybe we shouldn't remove this.

Comment (ala-ableton) [reply Dockerfile]: Addressed in 33fd174.

**Comment by ala-ableton** [comment]
@mst-ableton I've extended the commit message to explain the edge case better. Please have another look, thanks!

**Review by mst-ableton — APPROVED** [review]
Thanks for the detailed explanation! lgtm

**Pull request merged**